### ISSUE TITLE
guild view throws socket.gaierror: [Errno 8] nodename nor servname provided, or not known 

### ISSUE THREAD

**Issue by cfregly** [guildai/guildai#5]
```
$ guild check
guild_version:             0.3.0.dev19
guild_home:                /Users/cfregly/.guild
guild_install_location:    /Users/cfregly/miniconda3/lib/python3.6/site-packages/guild
installed_plugins:         cloudml, cpu, disk, gpu, keras, memory
python_version:            3.6.4 |Anaconda, Inc.| (default, Jan 16 2018, 12:04:33) [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
tensorflow_version:        1.6.0
tensorflow_cuda_support:   no
tensorflow_gpu_available:  no
tensorboard_version:       1.6.0
libcuda_version:           not loaded
libcudnn_version:          not loaded
nvidia_smi_version:        not installed
```
```
$ guild view
Running Guild View at http://cfregly:65025
Traceback (most recent call last):
  File "/Users/cfregly/miniconda3/bin/guild", line 11, in <module>
    sys.exit(main())
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/main_bootstrap.py", line 44, in main
    guild.main.main()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/main.py", line 30, in main
    _main()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/main.py", line 36, in _main
    guild.commands.main.main(standalone_mode=False)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 731, in __call__
    return self.main(*args, **kwargs)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 706, in main
    rv = self.invoke(ctx)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 1080, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 907, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 544, in invoke
    return callback(*args, **kwargs)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/click_util.py", line 140, in fn
    return fn0(*(args + (Args(kw),)))
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/view.py", line 45, in view
    view_impl.main(args)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/view_impl.py", line 283, in main
    view.serve_forever(data, host, port, args.no_open, args.dev)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/view.py", line 214, in serve_forever
    _serve_prod(data, host, port, no_open)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/view.py", line 236, in _serve_prod
    _start_view(data, host, port)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/view.py", line 241, in _start_view
    server = serving.make_server(host, port, app, threaded=True)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/werkzeug/serving.py", line 660, in make_server
    passthrough_errors, ssl_context, fd=fd)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/werkzeug/serving.py", line 577, in __init__
    self.address_family), handler)
  File "/Users/cfregly/miniconda3/lib/python3.6/socketserver.py", line 453, in __init__
    self.server_bind()
  File "/Users/cfregly/miniconda3/lib/python3.6/http/server.py", line 136, in server_bind
    socketserver.TCPServer.server_bind(self)
  File "/Users/cfregly/miniconda3/lib/python3.6/socketserver.py", line 467, in server_bind
    self.socket.bind(self.server_address)
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

**Comment by cfregly** [comment]
macos 10.13

**Comment by cfregly** [comment]
comments:  this is working for some people, but not me.

workaround from garrett is to added `--host=localhost` as follows:

```
guild view --host=localhost
```

**Comment by cfregly** [comment]
when using `--host=localhost` above, i can't get to tensorboard using "View in TensorBoard" link:

```
Traceback (most recent call last):
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/scripts/guild", line 30, in <module>
    main()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/scripts/guild", line 23, in main
    guild.main_bootstrap.main()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/main_bootstrap.py", line 44, in main
    guild.main.main()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/main.py", line 30, in main
    _main()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/main.py", line 36, in _main
    guild.commands.main.main(standalone_mode=False)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 731, in __call__
    return self.main(*args, **kwargs)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 706, in main
    rv = self.invoke(ctx)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 1080, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 907, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 544, in invoke
    return callback(*args, **kwargs)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/click_util.py", line 140, in fn
    return fn0(*(args + (Args(kw),)))
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/tensorboard.py", line 74, in tensorboard
    tensorboard_impl.main(args)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 90, in main
    monitor = RunsMonitor(logdir, args)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 50, in __init__
    self.run_once(exit_on_error=True)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 55, in run_once
    runs = self._runs()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 67, in _runs
    return runs_impl.runs_for_args(self.args)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 69, in runs_for_args
    filtered = filtered_runs(args, force_deleted)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 76, in filtered_runs
    filter=_runs_filter(args),
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 85, in _runs_filter
    _apply_cwd_guildfile_filter(args, filters)
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 92, in _apply_cwd_guildfile_filter
    cwd_guildfile = cmd_impl_support.cwd_guildfile()
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/cmd_impl_support.py", line 51, in cwd_guildfile
    import guild.model # expensive
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/model.py", line 28, in <module>
    from guild import guildfile
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/guildfile.py", line 27, in <module>
    from guild import resolver
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/resolver.py", line 26, in <module>
    from guild import pip_util
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/guild/pip_util.py", line 23, in <module>
    from pip.commands.download import DownloadCommand
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/pip/__init__.py", line 28, in <module>
    from pip.vcs import git, mercurial, subversion, bazaar  # noqa
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/pip/vcs/subversion.py", line 9, in <module>
    from pip.index import Link
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/pip/index.py", line 31, in <module>
    from pip.wheel import Wheel, wheel_ext
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/pip/wheel.py", line 6, in <module>
    import compileall
  File "/Users/cfregly/miniconda3/lib/python3.6/compileall.py", line 20, in <module>
    from concurrent.futures import ProcessPoolExecutor
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/concurrent/futures/__init__.py", line 8, in <module>
    from concurrent.futures._base import (FIRST_COMPLETED,
  File "/Users/cfregly/miniconda3/lib/python3.6/site-packages/concurrent/futures/_base.py", line 381
    raise exception_type, self._exception, self._traceback
                        ^
SyntaxError: invalid syntax
```

**Comment by BrentDorsey** [comment]
Selecting the "VIEW IN TENSORBOARD" link on the "Guild View" page raises the following error:

`/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from float to np.floating is deprecated. In future, it will be treated as np.float64 == np.dtype(float).type.
  from ._conv import register_converters as _register_converters
Traceback (most recent call last):
  File "/anaconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 106, in _load_guild_tensorboard_module
    from guild import tensorboard
  File "/anaconda3/lib/python3.6/site-packages/guild/tensorboard.py", line 51, in <module>
    from tensorboard.plugins.audio import audio_plugin
  File "/Users/bdorsey/.local/lib/python3.6/site-packages/tensorboard/plugins/audio/audio_plugin.py", line 27, in <module>
    from tensorboard import plugin_util
  File "/Users/bdorsey/.local/lib/python3.6/site-packages/tensorboard/plugin_util.py", line 24, in <module>
    import markdown
ModuleNotFoundError: No module named 'markdown'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/anaconda3/lib/python3.6/site-packages/guild/scripts/guild", line 30, in <module>
    main()
  File "/anaconda3/lib/python3.6/site-packages/guild/scripts/guild", line 23, in main
    guild.main_bootstrap.main()
  File "/anaconda3/lib/python3.6/site-packages/guild/main_bootstrap.py", line 44, in main
    guild.main.main()
  File "/anaconda3/lib/python3.6/site-packages/guild/main.py", line 30, in main
    _main()
  File "/anaconda3/lib/python3.6/site-packages/guild/main.py", line 36, in _main
    guild.commands.main.main(standalone_mode=False)
  File "/anaconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 731, in __call__
    return self.main(*args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 706, in main
    rv = self.invoke(ctx)
  File "/anaconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 1080, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/anaconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 907, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/anaconda3/lib/python3.6/site-packages/guild/external/click/core.py", line 544, in invoke
    return callback(*args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/guild/click_util.py", line 140, in fn
    return fn0(*(args + (Args(kw),)))
  File "/anaconda3/lib/python3.6/site-packages/guild/commands/tensorboard.py", line 74, in tensorboard
    tensorboard_impl.main(args)
  File "/anaconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 86, in main
    tensorboard = _load_guild_tensorboard_module()
  File "/anaconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 108, in _load_guild_tensorboard_module
    _handle_tensorboard_import_error(e)
  File "/anaconda3/lib/python3.6/site-packages/guild/commands/tensorboard_impl.py", line 113, in _handle_tensorboard_import_error
    if e.message == 'No module named tensorflow':
AttributeError: 'ModuleNotFoundError' object has no attribute 'message'`

**Comment by gar1t** [comment]
I'm going to move the "View in TensorBoard" info here to a new issue.

**Comment by gar1t** [comment]
@cfregly Your traceback is quite interesting - the source shows Python 2 syntax but the package is installed under Python 3.6 site packages. I'm not sure how that happens, but that's the problem there.

@BrentDorsey I moved your issue over to #9.

**Comment by gar1t** [comment]
Fixed in 0.3.1.dev1

**Comment by Naeemkh** [comment]
I had this exact problem on macOS. Users should add IP and host name into /etc/hosts file. I summarized it here:

https://stackoverflow.com/questions/69937085/socket-gethostbyname-fails-on-macos/69943923#69943923
